middleware/apimiddleware.go

```diff
@@ -229,13 +231,22 @@ func GetBlockHashCtx(r *http.Request) (string, error) {
 
 // GetAddressCtx retrieves the ctxAddress data from the request context. If not
 // set, the return value is an empty string.
-func GetAddressCtx(r *http.Request) string {
-	address, ok := r.Context().Value(CtxAddress).(string)
+func GetAddressCtx(r *http.Request, activeNetParams *chaincfg.Params) (string, error) {
+	addressStr, ok := r.Context().Value(CtxAddress).(string)
 	if !ok {
 		apiLog.Trace("address not set")
-		return ""
+		return "", fmt.Errorf("address not set")
+	}
+	address, err := dcrutil.DecodeAddress(addressStr)
+	if err != nil {
+		return "", fmt.Errorf("invalid address '%v': %v",
+			addressStr, err)
+	}
+	if !address.IsForNet(activeNetParams) {
+		return "", fmt.Errorf("%v is invalid for this network",
+			address)
 	}
-	return address
+	return address.String(), nil
```

Comment:
might as well return `addressStr` since we have it.

Reply:
addressStr it is.